labelbox/client.py

```diff
@@ -939,3 +941,136 @@ def get_model_run(self, model_run_id: str) -> ModelRun:
             A ModelRun object.
         """
         return self._get_single(Entity.ModelRun, model_run_id)
+
+    def assign_global_keys_to_data_rows(
+            self,
+            global_key_to_data_row_inputs: List[AssignGlobalKeyToDataRowInput],
+            timeout=30) -> List[Dict[str, str]]:
+        """
+        Assigns global keys to the related data rows.
+        
+        >>> global_key_data_row_inputs = [
+            AssignGlobalKeyToDataRowInput(data_row_id = "cl7asgri20yvo075b4vtfedjb", global_key = "key1"),
+            AssignGlobalKeyToDataRowInput(data_row_id = "cl7asgri10yvg075b4pz176ht", global_key = "key2")
+        ]
+        >>> client.assign_global_keys_to_data_rows(global_key_data_row_inputs)
+
+        Args:
+            A list of AssignGlobalKeyToDataRowInput objects.
+        Returns:
+            Returns successful assigned global keys and data rows
+        """
+
+        mutation_str = """mutation assignGlobalKeysToDataRowsPyApi($globalKeyDataRowLinks: [AssignGlobalKeyToDataRowInput!]!) { 
+            assignGlobalKeysToDataRows(data: {assignInputs: $globalKeyDataRowLinks}) { 
+                jobId 
+            }
+        }
+        """
+        mutation_params = {
+            'globalKeyDataRowLinks': [
+                input.dict(by_alias=True)
+                for input in global_key_to_data_row_inputs
+            ]
+        }
+        assign_global_keys_to_data_rows_job = self.execute(
+            mutation_str, mutation_params)
+
+        get_failed_assignments_str = """query getDataRowsForGlobalKeysPyApi($jobId: ID!) {
+            assignGlobalKeysToDataRowsResult(jobId: {id: $jobId}) {
+                jobStatus
+                data {
+                    sanitizedAssignments {
+                        dataRowId
+                        globalKey
+                    }
+                    invalidGlobalKeyAssignments {
+                        dataRowId
+                        globalKey
+                    }
+                    unmodifiedAssignments {
+                        dataRowId
+                        globalKey
+                    }
+                    accessDeniedAssignments {
+                        dataRowId
+                        globalKey
+                    }
+                }}}
+        """
+        get_failed_assignments_params = {
+            "jobId":
+                assign_global_keys_to_data_rows_job["assignGlobalKeysToDataRows"
+                                                   ]["jobId"]
+        }
+
+        while timeout >= 0:
```

Comment:
Can we use the same approach as `project.export_labels()`, using `time.time()` to calculate the timeout?
```
sleep_time = 2
start_time = time.time()
while True:
  # logic here
  current_time = time.time()
  if current_time - start_time > timeout_seconds:
     raise Exception()
  time.sleep(sleep_time)
```

I remember this approach is less error prone, and we've seen timeouts in less than `timeout_seconds` with the current approach